### PR TITLE
2019-10-07 Image Creation Process

### PR DESCRIPTION
**Pull request by cloudtoad**
A blog post about the current process for how images are created and presented to the user.

### CODE REVIEW

**Comment by Mierdin** [comment]
Is this still work in progress or ready for review?

File: content/post/2019-10-07-Image-Creation-Process.md [hunk 1]
```diff
@@ -0,0 +1,53 @@
+---
+date: 2019-10-07T00:00:00-00:00
+description: ""
+featuredimage: ""
```
Comment (Mierdin) [inline content/post/2019-10-07-Image-Creation-Process.md]: Please minimally add a featured image here. Would also be nice if there were some images inline to help with chunking, but having one here is important so it ties into our social automation, etc.

File: content/post/2019-10-07-Image-Creation-Process.md [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+tags: []
+title: "The Image Creation Process"
```
Comment (Mierdin) [inline content/post/2019-10-07-Image-Creation-Process.md]: To disambiguate, perhaps the below?

```suggestion
title: "The Antidote Endpoint Image Creation Process"
```

File: content/post/2019-10-07-Image-Creation-Process.md [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+
+In the case of applications like Asterisk PBX, Free-Range Routing, or Netbox images, we install or compile the relevant packages and dependencies.  On all images, we ensure that any APIs (including REST or Netconf) are enabled.
+
+Once we are done with the ISO image itself, we need to create a dockerfile for the image that launches it with QEMU.  The launch script for this container should look for and create the right network interfaces with QEMU, and stitch them to the underlying interfaces in the container.  It should also pass a mount-point to the virtual-machine so it can mount the lesson and configuration files.  Finally, the launch script should ensure the right ports are being forwarded from the underlying primary interface of the container (“eth0”) to the primary interface of the virtual-machine.
+
+The completed ISO files for QEMU are stored in a Google Cloud Platform bucket.  The docker images are stored in the “antidotelabs” docker hub repository.  Our base is done!
```
Comment (Mierdin) [inline content/post/2019-10-07-Image-Creation-Process.md]: This paragraph makes it sound like google cloud buckets are a build destination, and we're trying to only store unmodified disk images there. Maybe rephrase to mention that we store raw unmodified images (and are striving to do that for everything) in google cloud, but the only destination for this pipeline is docker hub - that would be more accurate.

File: content/post/2019-10-07-Image-Creation-Process.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+
+## Stage 2:  The Lesson Metafile
+
+Now that we have a base image, we need to start configuring it for a particular lesson at the time the lesson is launched.  This configuration starts with Syringe, the core component of the Antidote platform, reading the lesson metafile at the time a lesson is launched.
```
Comment (Mierdin) [inline content/post/2019-10-07-Image-Creation-Process.md]:
```suggestion
Now that we have a base image, we need to start configuring it for a particular lesson at the time the lesson is launched.  This configuration starts with Syringe, the core component of the Antidote platform, reading the lesson metafile (`lesson.meta.yaml`) at the time a lesson is launched.
```

File: content/post/2019-10-07-Image-Creation-Process.md [hunk 5]
```diff
@@ -0,0 +1,53 @@
...
+
+This Metafile describes which images to use for the lesson, and how the user will interact with those images.  For instance, the user might use SSH, NetConf, or a REST API to interact with the image.  
+
+Additionally, the metafile describes how the various virtual-machines in a lesson are interconnected, and Syringe will ensure that the right interfaces are created and connected with each other when the virtual-machines are launched.
+
+So far so good.  The base images are launched, we can access them, and they are wired together!
+
+## Stage 3:  Virtual-Machine Configuration
+
+The final step in preparing the image for the user is to configure the applications in it.  This can be done a variety of ways, including via NetConf, Python, or Ansible.  First, though, this is the stage where the hostname and interface IP addresses of the virtual-machine are configured.
```
Comment (Mierdin) [inline content/post/2019-10-07-Image-Creation-Process.md]:
```suggestion
The final step in preparing the image for the user is to configure the applications in it. [This can be done a variety of ways](https://antidoteproject.readthedocs.io/en/latest/platform/curricula/lessons/configuration.html), including via NetConf, Python, or Ansible.  First, though, this is the stage where the hostname and interface IP addresses of the virtual-machine are configured.  
```

File: content/post/2019-10-07-Image-Creation-Process.md (hunk 3 above)
Comment (cloudtoad) [reply content/post/2019-10-07-Image-Creation-Process.md]: I'll update the verbiage.

File: content/post/2019-10-07-Image-Creation-Process.md (hunk 1 above)
Comment (cloudtoad) [reply content/post/2019-10-07-Image-Creation-Process.md]: Working on image now.
